CI: TypeError: Multiple namespaces for array inputs

I'm seeing a new test failure in  #24041

The error is:

```
2025-11-23T07:04:07.8119600Z =================================== FAILURES ===================================
2025-11-23T07:04:07.8119858Z __________________ TestGetWindow.test_array_as_window[torch] ___________________
2025-11-23T07:04:07.8120014Z 
2025-11-23T07:04:07.8120084Z api_version = None, use_compat = None
2025-11-23T07:04:07.8120369Z xs = (array([0.00233883, 0.00527363, 0.00959968, 0.01563093, 0.0236949 ,
2025-11-23T07:04:07.8120575Z        0.03412321, 0.04724088, 0.06335451, 0.08273...111,
2025-11-23T07:04:07.8120873Z         112, 113, 114, 115, 116, 117, 118, 119, 120, 121, 122, 123, 124, 125,
2025-11-23T07:04:07.8121084Z         126, 127], device='cuda:0'))
2025-11-23T07:04:07.8121688Z namespaces = {<module 'scipy._lib.array_api_compat.torch' from '/home/admin/actions-runner/_work/scipy/scipy/build-cuda-install/usr.../_work/scipy/scipy/build-cuda-install/usr/lib/python3.13/site-packages/scipy/_lib/array_api_compat/numpy/__init__.py'>}
2025-11-23T07:04:07.8122413Z 
2025-11-23T07:04:07.8122491Z     def array_namespace(
2025-11-23T07:04:07.8122660Z         *xs: Array | complex | None,
2025-11-23T07:04:07.8122841Z         api_version: str | None = None,
2025-11-23T07:04:07.8123023Z         use_compat: bool | None = None,
2025-11-23T07:04:07.8123173Z     ) -> Namespace:
2025-11-23T07:04:07.8123295Z         """
2025-11-23T07:04:07.8123448Z         Get the array API compatible namespace for the arrays `xs`.
2025-11-23T07:04:07.8123651Z     
2025-11-23T07:04:07.8123790Z         Parameters
2025-11-23T07:04:07.8123920Z         ----------
2025-11-23T07:04:07.8124037Z         xs: arrays
2025-11-23T07:04:07.8124205Z             one or more arrays. xs can also be Python scalars (bool, int, float,
2025-11-23T07:04:07.8124419Z             complex, or None), which are ignored.
2025-11-23T07:04:07.8124573Z     
2025-11-23T07:04:07.8124676Z         api_version: str
2025-11-23T07:04:07.8124867Z             The newest version of the spec that you need support for (currently
2025-11-23T07:04:07.8125134Z             the compat library wrapped APIs support v2024.12).
2025-11-23T07:04:07.8125307Z     
2025-11-23T07:04:07.8125417Z         use_compat: bool or None
2025-11-23T07:04:07.8125616Z             If None (the default), the native namespace will be returned if it is
2025-11-23T07:04:07.8125877Z             already array API compatible, otherwise a compat wrapper is used. If
2025-11-23T07:04:07.8126131Z             True, the compat library wrapped library will be returned. If False,
2025-11-23T07:04:07.8126351Z             the native library namespace is returned.
2025-11-23T07:04:07.8126504Z     
2025-11-23T07:04:07.8126604Z         Returns
2025-11-23T07:04:07.8126717Z         -------
2025-11-23T07:04:07.8126825Z     
2025-11-23T07:04:07.8127005Z         out: namespace
2025-11-23T07:04:07.8127239Z             The array API compatible namespace corresponding to the arrays in `xs`.
2025-11-23T07:04:07.8127444Z     
2025-11-23T07:04:07.8127546Z         Raises
2025-11-23T07:04:07.8127660Z         ------
2025-11-23T07:04:07.8127767Z         TypeError
2025-11-23T07:04:07.8127937Z             If `xs` contains arrays from different array libraries or contains a
2025-11-23T07:04:07.8128126Z             non-array.
2025-11-23T07:04:07.8128244Z     
2025-11-23T07:04:07.8128333Z     
2025-11-23T07:04:07.8128505Z         Typical usage is to pass the arguments of a function to
2025-11-23T07:04:07.8128744Z         `array_namespace()` at the top of a function to get the corresponding
2025-11-23T07:04:07.8128951Z         array API namespace:
2025-11-23T07:04:07.8129075Z     
2025-11-23T07:04:07.8129186Z         .. code:: python
2025-11-23T07:04:07.8129316Z     
2025-11-23T07:04:07.8129420Z            def your_function(x, y):
2025-11-23T07:04:07.8129595Z                xp = array_api_compat.array_namespace(x, y)
2025-11-23T07:04:07.8129851Z                # Now use xp as the array library namespace
2025-11-23T07:04:07.8130042Z                return xp.mean(x, axis=0) + 2*xp.std(y, axis=0)
2025-11-23T07:04:07.8130210Z     
2025-11-23T07:04:07.8130301Z     
2025-11-23T07:04:07.8130459Z         Wrapped array namespaces can also be imported directly. For example,
2025-11-23T07:04:07.8130714Z         `array_namespace(np.array(...))` will return `array_api_compat.numpy`.
2025-11-23T07:04:07.8130961Z         This function will also work for any array library not wrapped by
2025-11-23T07:04:07.8131268Z         array-api-compat if it explicitly defines `__array_namespace__
2025-11-23T07:04:07.8131623Z         <https://data-apis.org/array-api/latest/API_specification/generated/array_api.array.__array_namespace__.html>`__
2025-11-23T07:04:07.8131939Z         (the wrapped namespace is always preferred if it exists).
2025-11-23T07:04:07.8132108Z     
2025-11-23T07:04:07.8132202Z         See Also
2025-11-23T07:04:07.8132317Z         --------
2025-11-23T07:04:07.8132419Z     
2025-11-23T07:04:07.8132523Z         is_array_api_obj
2025-11-23T07:04:07.8132644Z         is_numpy_array
2025-11-23T07:04:07.8132763Z         is_cupy_array
2025-11-23T07:04:07.8132882Z         is_torch_array
2025-11-23T07:04:07.8132993Z         is_dask_array
2025-11-23T07:04:07.8133125Z         is_jax_array
2025-11-23T07:04:07.8133243Z         is_pydata_sparse_array
2025-11-23T07:04:07.8133386Z     
2025-11-23T07:04:07.8133478Z         """
2025-11-23T07:04:07.8133598Z         namespaces: set[Namespace] = set()
2025-11-23T07:04:07.8133741Z         for x in xs:
2025-11-23T07:04:07.8133923Z             xp, info = _cls_to_namespace(cast(Hashable, type(x)), api_version, use_compat)
2025-11-23T07:04:07.8134135Z             if info is _ClsToXPInfo.SCALAR:
2025-11-23T07:04:07.8134286Z                 continue
2025-11-23T07:04:07.8134399Z     
2025-11-23T07:04:07.8134489Z             if (
2025-11-23T07:04:07.8134627Z                 info is _ClsToXPInfo.MAYBE_JAX_ZERO_GRADIENT
2025-11-23T07:04:07.8134801Z                 and _is_jax_zero_gradient_array(x)
2025-11-23T07:04:07.8134951Z             ):
2025-11-23T07:04:07.8135078Z                 xp = _jax_namespace(api_version, use_compat)
2025-11-23T07:04:07.8135273Z     
2025-11-23T07:04:07.8135369Z             if xp is None:
2025-11-23T07:04:07.8135521Z                 get_ns = getattr(x, "__array_namespace__", None)
2025-11-23T07:04:07.8135681Z                 if get_ns is None:
2025-11-23T07:04:07.8135875Z                     raise TypeError(f"{type(x).__name__} is not a supported array type")
2025-11-23T07:04:07.8136090Z                 if use_compat:
2025-11-23T07:04:07.8136223Z                     raise ValueError(
2025-11-23T07:04:07.8136420Z                         "The given array does not have an array-api-compat wrapper"
2025-11-23T07:04:07.8136600Z                     )
2025-11-23T07:04:07.8136732Z                 xp = get_ns(api_version=api_version)
2025-11-23T07:04:07.8136888Z     
2025-11-23T07:04:07.8137078Z             namespaces.add(xp)
2025-11-23T07:04:07.8137313Z     
2025-11-23T07:04:07.8137414Z         try:
2025-11-23T07:04:07.8137528Z >           (xp,) = namespaces
2025-11-23T07:04:07.8137655Z             ^^^^^
2025-11-23T07:04:07.8137816Z E           ValueError: too many values to unpack (expected 1)
2025-11-23T07:04:07.8137944Z 
2025-11-23T07:04:07.8137994Z api_version = None
2025-11-23T07:04:07.8138110Z info       = None
2025-11-23T07:04:07.8138641Z namespaces = {<module 'scipy._lib.array_api_compat.torch' from '/home/admin/actions-runner/_work/scipy/scipy/build-cuda-install/usr.../_work/scipy/scipy/build-cuda-install/usr/lib/python3.13/site-packages/scipy/_lib/array_api_compat/numpy/__init__.py'>}
2025-11-23T07:04:07.8139236Z use_compat = None
2025-11-23T07:04:07.8139389Z x          = tensor([  0,   1,   2,   3,   4,   5,   6,   7,   8,   9,  10,  11,  12,  13,
2025-11-23T07:04:07.8139574Z          14,  15,  16,  17,  18,  19,  2... 111,
2025-11-23T07:04:07.8139771Z         112, 113, 114, 115, 116, 117, 118, 119, 120, 121, 122, 123, 124, 125,
2025-11-23T07:04:07.8139963Z         126, 127], device='cuda:0')
2025-11-23T07:04:07.8140403Z xp         = <module 'scipy._lib.array_api_compat.torch' from '/home/admin/actions-runner/_work/scipy/scipy/build-cuda-install/usr/lib/python3.13/site-packages/scipy/_lib/array_api_compat/torch/__init__.py'>
2025-11-23T07:04:07.8140877Z xs         = (array([0.00233883, 0.00527363, 0.00959968, 0.01563093, 0.0236949 ,
2025-11-23T07:04:07.8141086Z        0.03412321, 0.04724088, 0.06335451, 0.08273...111,
2025-11-23T07:04:07.8141329Z         112, 113, 114, 115, 116, 117, 118, 119, 120, 121, 122, 123, 124, 125,
2025-11-23T07:04:07.8141497Z         126, 127], device='cuda:0'))
2025-11-23T07:04:07.8141593Z 
2025-11-23T07:04:07.8141683Z scipy/_lib/array_api_compat/common/_helpers.py:658: ValueError
2025-11-23T07:04:07.8141815Z 
2025-11-23T07:04:07.8141913Z During handling of the above exception, another exception occurred:
2025-11-23T07:04:07.8142060Z 
2025-11-23T07:04:07.8142182Z self = <scipy.signal.tests.test_windows.TestGetWindow object at 0x73088a170050>
2025-11-23T07:04:07.8142672Z xp = <module 'scipy._lib.array_api_compat.torch' from '/home/admin/actions-runner/_work/scipy/scipy/build-cuda-install/usr/lib/python3.13/site-packages/scipy/_lib/array_api_compat/torch/__init__.py'>
2025-11-23T07:04:07.8143040Z 
2025-11-23T07:04:07.8143098Z     @make_xp_test_case(windows.kaiser)
2025-11-23T07:04:07.8143254Z     def test_array_as_window(self, xp):
2025-11-23T07:04:07.8143397Z         # github issue 3603
2025-11-23T07:04:07.8143524Z         osfactor = 128
2025-11-23T07:04:07.8143645Z         sig = xp.arange(128)
2025-11-23T07:04:07.8143768Z     
2025-11-23T07:04:07.8143913Z         win = windows.get_window(('kaiser', 8.0), osfactor // 2, xp=xp)
2025-11-23T07:04:07.8144130Z         with assert_raises(ValueError, match='^window.shape='):
2025-11-23T07:04:07.8144360Z >           resample(sig, sig.shape[0] * osfactor, window=_xp_copy_to_numpy(win))
2025-11-23T07:04:07.8144506Z 
2025-11-23T07:04:07.8144551Z osfactor   = 128
2025-11-23T07:04:07.8144740Z self       = <scipy.signal.tests.test_windows.TestGetWindow object at 0x73088a170050>
2025-11-23T07:04:07.8144981Z sig        = tensor([  0,   1,   2,   3,   4,   5,   6,   7,   8,   9,  10,  11,  12,  13,
2025-11-23T07:04:07.8145214Z          14,  15,  16,  17,  18,  19,  2... 111,
2025-11-23T07:04:07.8145386Z         112, 113, 114, 115, 116, 117, 118, 119, 120, 121, 122, 123, 124, 125,
2025-11-23T07:04:07.8145552Z         126, 127], device='cuda:0')
2025-11-23T07:04:07.8145743Z win        = tensor([0.0023, 0.0053, 0.0096, 0.0156, 0.0237, 0.0341, 0.0472, 0.0634, 0.0827,
2025-11-23T07:04:07.8145938Z         0.1056, 0.1322, 0.1626, 0.1967...1626,
2025-11-23T07:04:07.8146116Z         0.1322, 0.1056, 0.0827, 0.0634, 0.0472, 0.0341, 0.0237, 0.0156, 0.0096,
2025-11-23T07:04:07.8146289Z         0.0053], device='cuda:0')
2025-11-23T07:04:07.8146728Z xp         = <module 'scipy._lib.array_api_compat.torch' from '/home/admin/actions-runner/_work/scipy/scipy/build-cuda-install/usr/lib/python3.13/site-packages/scipy/_lib/array_api_compat/torch/__init__.py'>
2025-11-23T07:04:07.8156398Z 
2025-11-23T07:04:07.8156534Z scipy/signal/tests/test_windows.py:919: 
2025-11-23T07:04:07.8157029Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2025-11-23T07:04:07.8157257Z scipy/signal/_support_alternative_backends.py:37: in wrapper
2025-11-23T07:04:07.8157455Z     xp = delegator(*args, **kwds)
2025-11-23T07:04:07.8157597Z          ^^^^^^^^^^^^^^^^^^^^^^^^
2025-11-23T07:04:07.8157889Z         args       = (tensor([  0,   1,   2,   3,   4,   5,   6,   7,   8,   9,  10,  11,  12,  13,
2025-11-23T07:04:07.8158113Z          14,  15,  16,  17,  18,  19,  ...      112, 113, 114, 115, 116, 117, 118, 119, 120, 121, 122, 123, 124, 125,
2025-11-23T07:04:07.8158338Z         126, 127], device='cuda:0'), 16384)
2025-11-23T07:04:07.8158528Z         delegator  = <function resample_signature at 0x73088d893880>
2025-11-23T07:04:07.8158733Z         func       = <function resample at 0x73088d95f920>
2025-11-23T07:04:07.8158951Z         kwds       = {'window': array([0.00233883, 0.00527363, 0.00959968, 0.01563093, 0.0236949 ,
2025-11-23T07:04:07.8159222Z        0.03412321, 0.04724088, 0.0633545...   0.10562887, 0.08273982, 0.06335451, 0.04724088, 0.03412321,
2025-11-23T07:04:07.8159450Z        0.0236949 , 0.01563093, 0.00959968, 0.00527363])}
2025-11-23T07:04:07.8159603Z         module_name = 'signal'
2025-11-23T07:04:07.8159770Z scipy/signal/_delegators.py:272: in resample_signature
2025-11-23T07:04:07.8160016Z     return array_namespace(x, t, _skip_if_str_or_tuple(window))
2025-11-23T07:04:07.8160202Z            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-11-23T07:04:07.8160344Z         axis       = 0
2025-11-23T07:04:07.8160460Z         domain     = 'time'
2025-11-23T07:04:07.8160585Z         num        = 16384
2025-11-23T07:04:07.8160702Z         t          = None
2025-11-23T07:04:07.8160874Z         window     = array([0.00233883, 0.00527363, 0.00959968, 0.01563093, 0.0236949 ,
2025-11-23T07:04:07.8161145Z        0.03412321, 0.04724088, 0.06335451, 0.082739...    0.10562887, 0.08273982, 0.06335451, 0.04724088, 0.03412321,
2025-11-23T07:04:09.4632932Z        0.0236949 , 0.01563093, 0.00959968, 0.00527363])
2025-11-23T07:04:09.4633520Z         x          = tensor([  0,   1,   2,   3,   4,   5,   6,   7,   8,   9,  10,  11,  12,  13,
2025-11-23T07:04:09.4633706Z          14,  15,  16,  17,  18,  19,  2... 111,
2025-11-23T07:04:09.4633900Z         112, 113, 114, 115, 116, 117, 118, 119, 120, 121, 122, 123, 124, 125,
2025-11-23T07:04:09.4634116Z         126, 127], device='cuda:0')
2025-11-23T07:04:09.4634304Z scipy/_lib/_array_api_override.py:150: in array_namespace
2025-11-23T07:04:09.4634548Z     return array_api_compat.array_namespace(*numpy_arrays, *api_arrays)
2025-11-23T07:04:09.4634756Z            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-11-23T07:04:09.4634974Z         api_arrays = [tensor([  0,   1,   2,   3,   4,   5,   6,   7,   8,   9,  10,  11,  12,  13,
2025-11-23T07:04:09.4635196Z          14,  15,  16,  17,  18,  19,  ...111,
2025-11-23T07:04:09.4635363Z         112, 113, 114, 115, 116, 117, 118, 119, 120, 121, 122, 123, 124, 125,
2025-11-23T07:04:09.4635702Z         126, 127], device='cuda:0')]
2025-11-23T07:04:09.4635853Z         arr_info   = <_ArrayClsInfo.numpy: 1>
2025-11-23T07:04:09.4636049Z         array      = array([0.00233883, 0.00527363, 0.00959968, 0.01563093, 0.0236949 ,
2025-11-23T07:04:09.4636422Z        0.03412321, 0.04724088, 0.06335451, 0.082739...    0.10562887, 0.08273982, 0.06335451, 0.04724088, 0.03412321,
2025-11-23T07:04:09.4636674Z        0.0236949 , 0.01563093, 0.00959968, 0.00527363])
2025-11-23T07:04:09.4637205Z         arrays     = (tensor([  0,   1,   2,   3,   4,   5,   6,   7,   8,   9,  10,  11,  12,  13,
2025-11-23T07:04:09.4637454Z          14,  15,  16,  17,  18,  19,  ...   0.10562887, 0.08273982, 0.06335451, 0.04724088, 0.03412321,
2025-11-23T07:04:09.4637695Z        0.0236949 , 0.01563093, 0.00959968, 0.00527363]))
2025-11-23T07:04:09.4637925Z         numpy_arrays = [array([0.00233883, 0.00527363, 0.00959968, 0.01563093, 0.0236949 ,
2025-11-23T07:04:09.4638219Z        0.03412321, 0.04724088, 0.06335451, 0.08273...   0.10562887, 0.08273982, 0.06335451, 0.04724088, 0.03412321,
2025-11-23T07:04:09.4638471Z        0.0236949 , 0.01563093, 0.00959968, 0.00527363])]
2025-11-23T07:04:09.4638653Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2025-11-23T07:04:09.4638814Z 
2025-11-23T07:04:09.4638877Z api_version = None, use_compat = None
2025-11-23T07:04:09.4639085Z xs = (array([0.00233883, 0.00527363, 0.00959968, 0.01563093, 0.0236949 ,
2025-11-23T07:04:09.4639275Z        0.03412321, 0.04724088, 0.06335451, 0.08273...111,
2025-11-23T07:04:09.4639456Z         112, 113, 114, 115, 116, 117, 118, 119, 120, 121, 122, 123, 124, 125,
2025-11-23T07:04:09.4639622Z         126, 127], device='cuda:0'))
2025-11-23T07:04:09.4640167Z namespaces = {<module 'scipy._lib.array_api_compat.torch' from '/home/admin/actions-runner/_work/scipy/scipy/build-cuda-install/usr.../_work/scipy/scipy/build-cuda-install/usr/lib/python3.13/site-packages/scipy/_lib/array_api_compat/numpy/__init__.py'>}
2025-11-23T07:04:09.4640659Z 
2025-11-23T07:04:09.4640716Z     def array_namespace(
2025-11-23T07:04:09.4640847Z         *xs: Array | complex | None,
2025-11-23T07:04:09.4640997Z         api_version: str | None = None,
2025-11-23T07:04:09.4641143Z         use_compat: bool | None = None,
2025-11-23T07:04:09.4641293Z     ) -> Namespace:
2025-11-23T07:04:09.4641405Z         """
2025-11-23T07:04:09.4641586Z         Get the array API compatible namespace for the arrays `xs`.
2025-11-23T07:04:09.4641762Z     
2025-11-23T07:04:09.4641858Z         Parameters
2025-11-23T07:04:09.4641980Z         ----------
2025-11-23T07:04:09.4642086Z         xs: arrays
2025-11-23T07:04:09.4642255Z             one or more arrays. xs can also be Python scalars (bool, int, float,
2025-11-23T07:04:09.4642459Z             complex, or None), which are ignored.
2025-11-23T07:04:09.4642607Z     
2025-11-23T07:04:09.4642706Z         api_version: str
2025-11-23T07:04:09.4642886Z             The newest version of the spec that you need support for (currently
2025-11-23T07:04:09.4643106Z             the compat library wrapped APIs support v2024.12).
2025-11-23T07:04:09.4643263Z     
2025-11-23T07:04:09.4643366Z         use_compat: bool or None
2025-11-23T07:04:09.4643547Z             If None (the default), the native namespace will be returned if it is
2025-11-23T07:04:09.4643794Z             already array API compatible, otherwise a compat wrapper is used. If
2025-11-23T07:04:09.4644032Z             True, the compat library wrapped library will be returned. If False,
2025-11-23T07:04:09.4644235Z             the native library namespace is returned.
2025-11-23T07:04:09.4644380Z     
2025-11-23T07:04:09.4644474Z         Returns
2025-11-23T07:04:09.4644583Z         -------
2025-11-23T07:04:09.4644681Z     
2025-11-23T07:04:09.4644777Z         out: namespace
2025-11-23T07:04:09.4644954Z             The array API compatible namespace corresponding to the arrays in `xs`.
2025-11-23T07:04:09.4645148Z     
2025-11-23T07:04:09.4645236Z         Raises
2025-11-23T07:04:09.4645369Z         ------
2025-11-23T07:04:09.4645469Z         TypeError
2025-11-23T07:04:09.4645630Z             If `xs` contains arrays from different array libraries or contains a
2025-11-23T07:04:09.4645812Z             non-array.
2025-11-23T07:04:09.4645921Z     
2025-11-23T07:04:09.4646010Z     
2025-11-23T07:04:09.4646149Z         Typical usage is to pass the arguments of a function to
2025-11-23T07:04:09.4646379Z         `array_namespace()` at the top of a function to get the corresponding
2025-11-23T07:04:09.4646571Z         array API namespace:
2025-11-23T07:04:09.4646695Z     
2025-11-23T07:04:09.4646807Z         .. code:: python
2025-11-23T07:04:09.4646993Z     
2025-11-23T07:04:09.4647101Z            def your_function(x, y):
2025-11-23T07:04:09.4647296Z                xp = array_api_compat.array_namespace(x, y)
2025-11-23T07:04:09.4647477Z                # Now use xp as the array library namespace
2025-11-23T07:04:09.4647668Z                return xp.mean(x, axis=0) + 2*xp.std(y, axis=0)
2025-11-23T07:04:09.4647833Z     
2025-11-23T07:04:09.4647933Z     
2025-11-23T07:04:09.4648088Z         Wrapped array namespaces can also be imported directly. For example,
2025-11-23T07:04:09.4648342Z         `array_namespace(np.array(...))` will return `array_api_compat.numpy`.
2025-11-23T07:04:09.4648618Z         This function will also work for any array library not wrapped by
2025-11-23T07:04:09.4648854Z         array-api-compat if it explicitly defines `__array_namespace__
2025-11-23T07:04:09.4649199Z         <https://data-apis.org/array-api/latest/API_specification/generated/array_api.array.__array_namespace__.html>`__
2025-11-23T07:04:09.4649510Z         (the wrapped namespace is always preferred if it exists).
2025-11-23T07:04:09.4649750Z     
2025-11-23T07:04:09.4649859Z         See Also
2025-11-23T07:04:09.4649967Z         --------
2025-11-23T07:04:09.4650075Z     
2025-11-23T07:04:09.4650177Z         is_array_api_obj
2025-11-23T07:04:09.4650301Z         is_numpy_array
2025-11-23T07:04:09.4650416Z         is_cupy_array
2025-11-23T07:04:09.4650531Z         is_torch_array
2025-11-23T07:04:09.4650640Z         is_dask_array
2025-11-23T07:04:09.4650752Z         is_jax_array
2025-11-23T07:04:09.4650868Z         is_pydata_sparse_array
2025-11-23T07:04:09.4650989Z     
2025-11-23T07:04:09.4651079Z         """
2025-11-23T07:04:09.4651224Z         namespaces: set[Namespace] = set()
2025-11-23T07:04:09.4651371Z         for x in xs:
2025-11-23T07:04:09.4651548Z             xp, info = _cls_to_namespace(cast(Hashable, type(x)), api_version, use_compat)
2025-11-23T07:04:09.4651766Z             if info is _ClsToXPInfo.SCALAR:
2025-11-23T07:04:09.4651910Z                 continue
2025-11-23T07:04:09.4652020Z     
2025-11-23T07:04:09.4652108Z             if (
2025-11-23T07:04:09.4652244Z                 info is _ClsToXPInfo.MAYBE_JAX_ZERO_GRADIENT
2025-11-23T07:04:09.4652416Z                 and _is_jax_zero_gradient_array(x)
2025-11-23T07:04:09.4652561Z             ):
2025-11-23T07:04:09.4652689Z                 xp = _jax_namespace(api_version, use_compat)
2025-11-23T07:04:09.4652834Z     
2025-11-23T07:04:09.4652933Z             if xp is None:
2025-11-23T07:04:09.4653076Z                 get_ns = getattr(x, "__array_namespace__", None)
2025-11-23T07:04:09.4653246Z                 if get_ns is None:
2025-11-23T07:04:09.4653436Z                     raise TypeError(f"{type(x).__name__} is not a supported array type")
2025-11-23T07:04:09.4653632Z                 if use_compat:
2025-11-23T07:04:09.4653762Z                     raise ValueError(
2025-11-23T07:04:09.4653952Z                         "The given array does not have an array-api-compat wrapper"
2025-11-23T07:04:09.4654130Z                     )
2025-11-23T07:04:09.4654259Z                 xp = get_ns(api_version=api_version)
2025-11-23T07:04:09.4654402Z     
2025-11-23T07:04:09.4654503Z             namespaces.add(xp)
2025-11-23T07:04:09.4654630Z     
2025-11-23T07:04:09.4654721Z         try:
2025-11-23T07:04:09.4654829Z             (xp,) = namespaces
2025-11-23T07:04:09.4654979Z             return xp
2025-11-23T07:04:09.4655098Z         except ValueError:
2025-11-23T07:04:09.4655221Z             if not namespaces:
2025-11-23T07:04:09.4655351Z                 raise TypeError(
2025-11-23T07:04:09.4655532Z                     "array_namespace requires at least one non-scalar array input"
2025-11-23T07:04:09.4655718Z                 )
2025-11-23T07:04:09.4655895Z >           raise TypeError(f"Multiple namespaces for array inputs: {namespaces}")
2025-11-23T07:04:09.4656839Z E           TypeError: Multiple namespaces for array inputs: {<module 'scipy._lib.array_api_compat.torch' from '/home/admin/actions-runner/_work/scipy/scipy/build-cuda-install/usr/lib/python3.13/site-packages/scipy/_lib/array_api_compat/torch/__init__.py'>, <module 'scipy._lib.array_api_compat.numpy' from '/home/admin/actions-runner/_work/scipy/scipy/build-cuda-install/usr/lib/python3.13/site-packages/scipy/_lib/array_api_compat/numpy/__init__.py'>}
2025-11-23T07:04:09.4657867Z 
2025-11-23T07:04:09.4657933Z api_version = None
2025-11-23T07:04:09.4658079Z info       = None
2025-11-23T07:04:09.4658655Z namespaces = {<module 'scipy._lib.array_api_compat.torch' from '/home/admin/actions-runner/_work/scipy/scipy/build-cuda-install/usr.../_work/scipy/scipy/build-cuda-install/usr/lib/python3.13/site-packages/scipy/_lib/array_api_compat/numpy/__init__.py'>}
2025-11-23T07:04:09.4659206Z use_compat = None
2025-11-23T07:04:09.4659346Z x          = tensor([  0,   1,   2,   3,   4,   5,   6,   7,   8,   9,  10,  11,  12,  13,
2025-11-23T07:04:09.4659525Z          14,  15,  16,  17,  18,  19,  2... 111,
2025-11-23T07:04:09.4659697Z         112, 113, 114, 115, 116, 117, 118, 119, 120, 121, 122, 123, 124, 125,
2025-11-23T07:04:09.4659861Z         126, 127], device='cuda:0')
2025-11-23T07:04:09.4660270Z xp         = <module 'scipy._lib.array_api_compat.torch' from '/home/admin/actions-runner/_work/scipy/scipy/build-cuda-install/usr/lib/python3.13/site-packages/scipy/_lib/array_api_compat/torch/__init__.py'>
2025-11-23T07:04:09.4660709Z xs         = (array([0.00233883, 0.00527363, 0.00959968, 0.01563093, 0.0236949 ,
2025-11-23T07:04:09.4660899Z        0.03412321, 0.04724088, 0.06335451, 0.08273...111,
2025-11-23T07:04:09.4661072Z         112, 113, 114, 115, 116, 117, 118, 119, 120, 121, 122, 123, 124, 125,
2025-11-23T07:04:09.4661239Z         126, 127], device='cuda:0'))
2025-11-23T07:04:09.4661349Z 
2025-11-23T07:04:09.4661442Z scipy/_lib/array_api_compat/common/_helpers.py:665: TypeError
2025-11-23T07:04:09.4661641Z =========================== short test summary info ============================
2025-11-23T07:04:09.4662643Z FAILED scipy/signal/tests/test_windows.py::TestGetWindow::test_array_as_window[torch] - TypeError: Multiple namespaces for array inputs: {<module 'scipy._lib.array_api_compat.torch' from '/home/admin/actions-runner/_work/scipy/scipy/build-cuda-install/usr/lib/python3.13/site-packages/scipy/_lib/array_api_compat/torch/__init__.py'>, <module 'scipy._lib.array_api_compat.numpy' from '/home/admin/actions-runner/_work/scipy/scipy/build-cuda-install/usr/lib/python3.13/site-packages/scipy/_lib/array_api_compat/numpy/__init__.py'>}
2025-11-23T07:04:09.4663673Z = 1 failed, 6809 passed, 12570 skipped, 67835 deselected, 22 xfailed in 93.78s (0:01:33) =
2025-11-23T07:04:11.3991463Z ##[error]Process completed with exit code 1.
```

Not sure what causes it.

I see a run 10 hours ago which succeeds: https://github.com/scipy/scipy/actions/runs/19601037137/job/56132575128

And one 9 hours ago which fails: https://github.com/scipy/scipy/actions/runs/19601277275/job/56133165941

--- a/scipy/signal/tests/test_windows.py
+++ b/scipy/signal/tests/test_windows.py
@@ -915,8 +915,9 @@ class TestGetWindow:
         sig = xp.arange(128)
 
         win = windows.get_window(('kaiser', 8.0), osfactor // 2, xp=xp)
-        with assert_raises(ValueError, match='^window.shape='):
-            resample(sig, sig.shape[0] * osfactor, window=_xp_copy_to_numpy(win))
+        mesg = "^window must" if is_cupy(xp) else "^window.shape="
+        with assert_raises(ValueError, match=mesg):
+            resample(sig, sig.shape[0] * osfactor, window=win)
 
     @make_xp_test_case(windows.general_cosine)
     def test_general_cosine(self, xp):

